### ISSUE TITLE
Specify and use local image repository in the yaml to create a galasa ecosystem

### ISSUE THREAD

**Issue by SteamboatHarry** [galasa-dev/projectmanagement#666]
When creating a Galasa Ecosystem there is a property "dockerRegistry" to tell where to pull docker images from. If I fill in my local image repository it looks like it's not used. 
For example: I get an error "Failed to pull image quay.io/coreos/etcd:v3.4.3" but I expected a pull image "[local image repository endpoint]/quay.io/coreos/etcd:v3.4.3" or something like that.

I need to have the images in a local repository because I can't access (not directly or via proxy) the original repositories.

**Comment by Michael-Baylis** [comment]
ASAP please James,     need to provide the repos for the etcd and couchdb images in the config,  same for rest.   Possibly have one config parameter to override all external images.

**Comment by Michael-Baylis** [comment]
Actually,  for every deployment, pod, statefulset, job,  you need to be able to override the image,  so we can mix and match versions repos etc,   incase we need to back out something.  In the docker operator, we can override the image name for each server instance.

**Comment by Jimbo4794** [comment]
Yes I agree with this Mike. Im also thinking with the wide variations between different docker repo's im thinking the best approach is to have the CRD for each resource able to override the image name (and version), rather than just provding a repo.  Should look something like this: 
```
...
monitoring:
    nodeSelector:
      beta.kubernetes.io/arch: amd64
    // Grafana
    grafanaReplicas: 1
    grafanaStorage: "200Mi"
    grafanaImageName: "grafana/grafana"
    grafanaImageVersion: "latest"
    
    // Prometheus
    prometheusReplicas: 1
    prometheusStorage: "200Mi"
    prometheusImageName: "prom/prometheus"
    prometheusImageVersion: "v2.10.0"
    
    // Metrics
    metricsReplicas: 1
    // No version number as it pulls from the galasa version
    metricsImageName: "docker.galasa.dev/galasa-boot-embedded-amd64"
...    
```
This was the image can be pull from a variey for repos, with the naming structure relevant for that type of repo (like whether it has namespaces or jsut raw images for example)

**Comment by Jimbo4794** [comment]
Changed my mind on the Galasa version. Going to make it so it can be specified per component

**Comment by Jimbo4794** [comment]
PR ready for review: https://github.com/galasa-dev/extensions/pull/141

**Comment by Jimbo4794** [comment]
Merged. 

@SteamboatHarry you will now need to pull the 0.15.0-SNAPSHOT operator from docker.galasa.dev to get the changes. You will also need to re-apply the CRD definition as i have added the image and version components. I have updated the basic CR given in the repo, but there isnt any doc yet on all the configurations. For the time being, the names for the CR YAML can be found: https://github.com/galasa-dev/extensions/blob/master/galasa-ecosystem-kubernetes-operator/pkg/apis/galasa/v1alpha1/galasaecosystem_types.go

For example: 
```
        // +kubebuilder:default="v2.10.0"
	PrometheusImageVersion string `json:"prometheusImageVersion,omitempty"`
```
Refers to the yaml component `prometheusImageVersion` which doesnt have to be specified (and will use the default if not)
